graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/method/MethodDataFetcherFactory.kt

```diff
@@ -29,12 +34,53 @@ import java.lang.reflect.Method
  * Resolving of method arguments is handled by the supplied [argument resolvers][ArgumentResolver].
  */
 class MethodDataFetcherFactory(
-    argumentResolvers: List<ArgumentResolver>,
+    private val argumentResolvers: List<ArgumentResolver>,
     private val parameterNameDiscoverer: ParameterNameDiscoverer = DefaultParameterNameDiscoverer()
 ) {
 
     private val resolvers = ArgumentResolverComposite(argumentResolvers)
 
+    fun checkInputArgumentsAreValid(method: Method, argumentNames: Set<String>) {
+        val inputArgumentResolvers = argumentResolvers.filterIsInstance<AbstractInputArgumentResolver>()
+        if (inputArgumentResolvers.isEmpty()) return
+
+        val bridgedMethod: Method = BridgeMethodResolver.findBridgedMethod(method)
+        val methodParameters: List<MethodParameter> = bridgedMethod.parameters.map { parameter ->
+            val methodParameter = SynthesizingMethodParameter.forParameter(parameter)
+            methodParameter.initParameterNameDiscovery(parameterNameDiscoverer)
+            methodParameter
+        }
+        val annotatedMethodParams = methodParameters.filter { it.hasParameterAnnotation(InputArgument::class.java) }
+        annotatedMethodParams.forEach { m ->
+            val selectedArgResolver = resolvers.getArgumentResolver(m)
+                ?: throw DataFetcherInputArgumentSchemaMismatchException("No suitable argument resolver found for method `${m.parameterName} on ${m.declaringClass}`.")
+            if (selectedArgResolver is AbstractInputArgumentResolver) {
+                val argName = selectedArgResolver.resolveArgumentName(m) ?: return@forEach
+                if (!argumentNames.contains(argName)) {
+                    val paramName = m.parameterName ?: return@forEach
+                    val arguments = if (argumentNames.isNotEmpty()) {
+                        "Found the following argument(s) in the schema: " + argumentNames.joinToString(prefix = "[", postfix = "]")
+                    } else {
+                        "No arguments on the field are defined in the schema."
+                    }
+
+                    when (selectedArgResolver) {
+                        is InputArgumentResolver -> throw DataFetcherInputArgumentSchemaMismatchException(
+                            "@InputArgument(name = \"$argName\") defined in ${method.declaringClass} in method `${method.name}` " +
+                                "on parameter named `$paramName` has no matching argument with name `$argName` in the GraphQL schema. " +
+                                arguments
+                        )
+                        is FallbackEnvironmentArgumentResolver -> throw DataFetcherInputArgumentSchemaMismatchException(
```

Comment:
Kind of related to my other comment. Is there a test that actually exercises this case? I don't know that it's possible, since we are only checking parameters that are annotated with `@InputArgument`, so basically by definition the resolver would always be `InputArgumentResolver`. IMO it isn't necessary to check the resolvers, we could simply validate that the field name from the annotation matches something from the schema, and I don't know that it needs to live here.